### ISSUE TITLE
Unable to parse packets from an EMQ X broker with QoS > 0

### ISSUE THREAD

**Issue by MrcJkb** [dustin/mqtt-hs#31]
## How to reproduce
* Run the [emqx 4.4.4 docker container](https://hub.docker.com/_/emqx)
* Modify the URI in `app/example/Main.hs` so that the client connects to `mqtt://localhost:1883`.
* Run the example.

### Expected behaviour
The same as when using the mosquitto broker.

### Actual behaviour
The client disconnects after publishing the first message with QoS > 0

### Notes

It appears that `parsePacket` (`Network.MQTT.Types`) results in 
```
ParseError {errorContexts = ["240"], errorMessage = "Failed reading: satisfy", errorPosition = 1:90 (89)}
```
(when called from `doSrc` in `Network.MQTT.Client`) causing the client to disconnect.
Adding an `UnknownPkt` constructor prevents the error (but doesn't solve the problem, of course).

I'm not yet sure if EMQ X is sending bad packets, or if the issue is on `net-mqtt`'s side...

**Comment by dustin** [comment]
Can you give me a pcap?

**Comment by dustin** [comment]
Actually, I can give this a go.  I just remember I got my thinkpad back recently so I should be able to run the docker image.

**Comment by dustin** [comment]
Unfortunately, this worked for me, so I may need that pcap after all.

**Comment by MrcJkb** [comment]
Strange. I just tried with emqx 5.0.3 (on NixOS) and got the same behaviour.

Anyways, here are the pcaps. I commented out the async publishing and ran one with QoS1 and one with QoS2.
[net-mqtt-pcaps.tar.gz](https://github.com/dustin/mqtt-hs/files/9148487/net-mqtt-pcaps.tar.gz)

**Comment by dustin** [comment]
OK, cool.  I was able to figure this out.  Thanks for the report and all the help!  It was a little tricky.

**Comment by dustin** [comment]
I published a new release, [0.8.2.2](https://hackage.haskell.org/package/net-mqtt-0.8.2.2) with this fix.

**Comment by MrcJkb** [comment]
Wow, that was fast. I just gave it a test run and it works.
Thanks!
